ddl/backfilling_operator.go

```diff
@@ -141,6 +147,60 @@ func NewAddIndexIngestPipeline(
 	), nil
 }
 
+// NewWriteIndexToExternalStoragePipeline creates a pipeline for writing index to external storage.
+func NewWriteIndexToExternalStoragePipeline(
+	ctx *OperatorCtx,
+	store kv.Storage,
+	sessPool opSessPool,
+	sessCtx sessionctx.Context,
+	jobID int64,
+	subtaskID int64,
+	tbl table.PhysicalTable,
+	idxInfo *model.IndexInfo,
+	startKey, endKey kv.Key,
+	totalRowCount *atomic.Int64,
+	onClose external.OnCloseFunc,
+) (*operator.AsyncPipeline, error) {
+	index := tables.NewIndex(tbl.GetPhysicalID(), tbl.Meta(), idxInfo)
+	copCtx, err := NewCopContext(tbl.Meta(), idxInfo, sessCtx)
+	if err != nil {
+		return nil, err
+	}
+	poolSize := copReadChunkPoolSize()
+	srcChkPool := make(chan *chunk.Chunk, poolSize)
+	for i := 0; i < poolSize; i++ {
+		srcChkPool <- chunk.NewChunkWithCapacity(copCtx.fieldTps, copReadBatchSize())
+	}
+	readerCnt := int(variable.GetDDLReorgWorkerCounter())
+	writerCnt := 1
+
+	// TODO(tangenta): replace uri with global variable.
```

Comment:
and we should store it in task meta on creation, as we don't forbid user alter it during subtask execution.